lib/rules/images-no-direct-imports.js

```diff
@@ -0,0 +1,49 @@
+const {basename, dirname, extname} = require('path');
+const resolve = require('eslint-module-utils/resolve').default;
+const {docsUrl} = require('../utilities');
+
+function isImageImport(resolvedSource) {
+  return /\.(svg|png|jpg)$/.test(resolvedSource);
+}
+
+function isImportFromCurrentFolderIndex(contextFilename, resolvedSource) {
+  const isIndex =
+    basename(contextFilename, extname(contextFilename)) === 'index';
+
+  return isIndex && dirname(resolvedSource) === dirname(contextFilename);
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        "Prefer that imports of image files go through an index file within the image's directory instead of directly referencing images within a component.",
+      category: 'Best Practices',
+      recommended: false,
+      uri: docsUrl('images-no-direct-imports'),
+    },
+    fixable: null,
+  },
+  create(context) {
+    function checkNode(node) {
+      const resolvedSource = resolve(node.source.value, context);
+
+      if (
+        resolvedSource &&
+        isImageImport(resolvedSource) &&
+        !isImportFromCurrentFolderIndex(context.getFilename(), resolvedSource)
+      ) {
+        context.report({
+          node,
+          message:
+            'Do not import image files directly, instead import from the index of the folder containing the image',
```

Comment:
You are explaining this rule in a few different ways/different wordings. It would be more clear to stick to one.